File: app/src/main/java/com/ardev/builder/project/impl/ModuleImpl.java
package com.ardev.builder.project.impl;

import androidx.annotation.Nullable;

import com.ardev.builder.project.api.FileManager;
import com.ardev.builder.project.api.Module;
import java.io.File;
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class ModuleImpl implements Module {

    /**
     * Concurrent writes to this field are via CASes only, using the {@link #updater}
     */
    
    private final File mRoot;
    
    private FileManager mFileManager;

    public ModuleImpl(File root) {
        mRoot = root;
        mFileManager = new FileManagerImpl(root);
    }

    @Override
    public void open() throws IOException {
        if(!getBuildDirectory().exists()) getBuildDirectory().mkdir();
        if(!getCacheDirectory().exists()) getCacheDirectory().mkdir();
        
    }

    @Override
    public void clear() {

    }

    @Override
    public void index() {

    }

    @Override
    public File getBuildDirectory() {
    return new File(getRootFile(), "build");
   
    }
	
	@Override
    public File getCacheDirectory() {
    return new File(getBuildDirectory(), ".cache");
    }

  

    @Override
    public FileManager getFileManager() {
        return mFileManager;
    }

    @Override
    public File getRootFile() {
        return mRoot;
    }

    
    

    @Override
    public int hashCode() {
        return mRoot.hashCode();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ModuleImpl)) return false;
        ModuleImpl project = (ModuleImpl) o;
        return mRoot.equals(project.mRoot);
    }

   
}

File: app/src/main/java/com/ardev/builder/project/impl/ModuleImpl.kt
package com.ardev.builder.project.impl

import com.ardev.builder.project.api.Module
import java.io.File
import java.io.IOException

class ModuleImpl(private val root: File) : Module {
    
    override val rootFile: File = root
    
    override val libsDir: File = File(rootFile, "libs")
    
    override val buildDir: File = File(rootFile, "build")
    
    override val cacheDir: File = File(buildDir, ".cache")
               
    override fun open() {    
       resolveDirs(buildDir, cacheDir)
        
        
    }

    override fun clear() {
     
    }

    override fun index() {
        
    }
                        
    override fun hashCode(): Int {
        return root.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ModuleImpl) return false
        return root == other.root
    }
}
